react/components/Promotion/EffectsSection/PriceForm/PriceTableEffect.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import { Radio, EXPERIMENTAL_Select as Select } from 'vtex.styleguide'
+import { useIntl } from 'react-intl'
+import { useQuery } from 'react-apollo'
+
+import getPriceTables from '../../../../graphql/getPriceTables.graphql'
+import getSalesChannels from '../../../../graphql/getSalesChannels.graphql'
+import { SelectValue } from '../../Conditions/Objects/SelectObject'
+import { PriceEffect } from '../../../../utils/promotion/typings'
+
+type PriceTableEffectProps = {
+  priceEffect: PriceEffect
+  onSelect: (discountType: string) => void
+  onChangeDiscount: (priceTableId: string) => void
+}
+
+const PriceTableEffect: React.FC<PriceTableEffectProps> = ({
+  priceEffect,
+  onSelect,
+  onChangeDiscount,
+}) => {
+  const intl = useIntl()
+  const isSelected = priceEffect.discountType === 'priceTable'
+
+  const { loading: loadingPriceTables, data: priceTablesResponse } = useQuery(
+    getPriceTables,
+    {
+      skip: !isSelected,
+    }
+  )
+  const {
+    loading: loadingSalesChannels,
+    data: salesChannelsResponse,
+  } = useQuery(getSalesChannels, {
+    skip: !isSelected,
+  })
+
+  const salesChannelsIds =
+    salesChannelsResponse?.getSalesChannels.map(
+      (sc: { id: string }) => sc.id
```

Comment:
```suggestion
      ({ id }) => id
```
Also, this should already be correctly typed.

Reply:
We are actually using `useQuery` without typing it all the time 😞Gonna type it.